### ISSUE TITLE
Dimension need double-check. Results are differenct when using caffemodel and torchmodel(with caffe weight)

### ISSUE THREAD

**Issue by tracer9** [qijiezhao/pseudo-3d-pytorch#11]
Hi Qijie:
I really appreciate your work. Recently, I am focus on video summarization and I want to copy P3D caffemodel into PyTorch one.

I just found that given same input video clip, the result **varied largely** when using caffe model and pytorch model with caffe weights. see below.
![image](https://user-images.githubusercontent.com/6302380/34756574-d2f4ca00-f606-11e7-996f-d9c35566ab2e.png)
(This is caffe result, blob['fc_kinetic'])

![image](https://user-images.githubusercontent.com/6302380/34756595-f49f9eb4-f606-11e7-9be3-f4c07ed63866.png)
(This is pytorch result)

I did some analysis myself and I found that after the first pooling, **the feature map size** in your code is `(1, 64, 8, 39, 39)` while in caffe, `(1, 64, 8, 40, 40)`.

I re-check the pooling param setting both in your code and official prototxt and found no difference. Then I realize that this may be because pooling implementation is differenct in caffe and pytorch.

See:
![image](https://user-images.githubusercontent.com/6302380/34756685-8ab165c2-f607-11e7-89f9-d7404ad5baeb.png)

![image](https://user-images.githubusercontent.com/6302380/34756706-aac2227a-f607-11e7-827c-c858daea888a.png)

And I do observe tiny difference after the first pooling layer.

I also attach my convert script file, If possible, could you help me to figure it out why the result is so different?

With my sincere thanks.

[convert.py.txt](https://github.com/qijiezhao/pseudo-3d-pytorch/files/1617585/convert.py.txt)

**Comment by qijiezhao** [comment]
It's mainly caused by the implementation of the first group of ST structure when time-dimension is 16. The only difference between the authors' and mine is the setting of padding. However, this would not bring influence to fine-tuning situation. If anyone who wanted to directly use the weights, I advise to change it as the authors'.

**Comment by qijiezhao** [comment]
I am going to finetune a modified version of 2+1 structure based new model in the coming anet18 competition.  I believe this would bring you more convenience for extracting high-level semantics.
